packages/core/upload/server/services/upload.js

```diff
@@ -106,7 +108,7 @@ module.exports = ({ strapi }) => ({
     const currentFile = this.formatFileInfo(
       {
         filename: file.name,
-        type: file.type,
+        type: file.type || mime.lookup(file.name) || 'application/octet-stream',
```

Comment:
@petersg83 @alexandrebodin To fix the issue mentioned [here](https://github.com/strapi/strapi/issues/12897#issuecomment-1117721915) I had to introduce the check also in this place because calling the API programmatically doesn't give us the koa context:

https://github.com/strapi/strapi/blob/39047e9f8476c435d12f5bf58d664d925a9adb35/packages/core/upload/server/services/upload.js#L375-L388

However: `enhanceFile` is called in many places already and I'm wondering if it would make sense to move the logic from `packages/core/strapi/lib/middlewares/body.js` in there and to make sure it is called every time a file is uploaded ...?

Reply:
From what I understood the issue you mentioned may be already fixed by the code in the middleware as the user probably uses the `files` uploaded though the API. We can avoid modifying the upload service (as it requires more discussion I think) and ask the users if the change in the middleware already fixes their issue? wdyt ?

Reply:
Just to update here-- I used `type` instead of `mime` and everything works like a charm!
I assumed that the column name of `mime` is the same as the property passed to the upload method

Thank you for all your help

Reply:
Yes but if you call `strapi.entityService.uploadFiles()` with files that are coming from the API request, then the files have already been mutated because they went through koa.